pkg/monitor/controller_test.go

```diff
@@ -0,0 +1,649 @@
+			Name:      "TestExternalNode",
+			Namespace: "ns1",
+			Labels:    map[string]string{"en": "vm2"},
+		},
+	}
+	tc := []struct {
+		name                 string
+		key                  string
+		existingExternalNode *v1alpha1.ExternalNode
+		expectedAgentCRD     *v1beta1.AntreaAgentInfo
+		expectedError        string
+	}{
+		{
+			name:          "Invalid key format",
+			key:           "namespace/name/error",
+			expectedError: "unexpected key format: \"namespace/name/error\"",
+		},
+		{
+			name:          "Key does not exist",
+			key:           "default/vm2-e8be5",
+			expectedError: "",
+		},
+		{
+			name:                 "key exists",
+			key:                  "ns1/TestExternalNode",
+			existingExternalNode: en,
+			expectedAgentCRD: &v1beta1.AntreaAgentInfo{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "TestExternalNode",
+					ResourceVersion: "",
+					Generation:      0,
+				},
+				NetworkPolicyControllerInfo: v1beta1.NetworkPolicyControllerInfo{
+					NetworkPolicyNum:  0,
+					AddressGroupNum:   0,
+					AppliedToGroupNum: 0,
+				},
+			},
+			expectedError: "",
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			clientset := fakeclientset.NewSimpleClientset()
+			if tt.existingExternalNode != nil {
+				clientset = fakeclientset.NewSimpleClientset(tt.existingExternalNode)
+			}
+			controller := newControllerMonitor(clientset)
+			initController(controller)
+			err := controller.controllerMonitor.syncExternalNode(tt.key)
+			if tt.expectedError != "" {
+				assert.Equal(t, tt.expectedError, err.Error())
+			} else {
+				assert.Nil(t, err)
+				if tt.expectedAgentCRD != nil {
+					crd, err := controller.controllerMonitor.client.CrdV1beta1().AntreaAgentInfos().Get(context.TODO(), tt.expectedAgentCRD.Name, metav1.GetOptions{})
+					assert.Nil(t, err)
+					assert.Equal(t, tt.expectedAgentCRD, crd)
+				}
+			}
+		})
+	}
+}
+
+func TestSyncNode(t *testing.T) {
+	tc := []struct {
+		name             string
+		key              string
+		expectedError    string
+		existingNode     *v1.Node
+		expectedAgentCRD *v1beta1.AntreaAgentInfo
+	}{
+		{
+			name:          "Invalid key format",
+			key:           "namespace/name/error",
+			expectedError: "unexpected key format: \"namespace/name/error\"",
+		},
+		{
+			name:          "Key does not exist",
+			key:           "default/vm2-e8be5",
+			expectedError: "",
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			clientset := fakeclientset.NewSimpleClientset()
+			controller := newControllerMonitor(clientset)
+			initController(controller)
+			err := controller.controllerMonitor.syncNode(tt.key)
+			if tt.expectedError != "" {
+				assert.Equal(t, tt.expectedError, err.Error())
+			} else {
+				assert.Nil(t, err)
+				crd, _ := controller.controllerMonitor.client.CrdV1beta1().AntreaAgentInfos().Get(context.TODO(), "TestExternalNode", metav1.GetOptions{})
+				assert.Equal(t, tt.expectedAgentCRD, crd)
```

Comment:
I didn't see any case has `tt.expectedAgentCRD` value. I suppose you need one more case to cover it?

Reply:
Added required test case